lib/vanagon/platform/windows.rb

```diff
@@ -67,7 +67,39 @@ def generate_nuget_package(project)
         "cp $(tempdir)/#{project.name}/#{project.name}-#{@architecture}.#{nuget_package_version(project.version, project.release)}.nupkg ./output/#{target_dir}/#{nuget_package_name(project)}"]
       end
 
-      # Method to derive the package name for the project.
+      # The specific bits used to generate a windows msi package for a given project
+      # [TBD - More to add]
+      #
+      # @param project [Vanagon::Project] project to build a msi package of
+      # @return [Array] list of commands required to build an msi package for the given project from a tarball
+      def generate_msi_package(project)
+        target_dir = project.repo ? output_dir(project.repo) : output_dir
+        ["mkdir -p output/#{target_dir}",
+        "mkdir -p $(tempdir)/STAGING",
+        # need to do things here.
+        "cp file-list $(tempdir)/#{project.name}/tools/file-list.txt", # May not need this (nuget only)
+        # this tarball contains programfiles/puppetlabs, Programdata etc.
+        # Its an image of where the files are going on the system.
+        # tempdir/PROGRAMFILES
+        # tempdir/PROGRAMDATA
+        "gunzip -c #{project.name}-#{project.version}.tar.gz | '#{@tar}' -C '$(tempdir)/#{project.name}/tools' --strip-components 1 -xf -",
+        # We can iterate through the complete directory set here in one operation instead of dealing with each "puppet agent component" seperately as in the current Rakefile.
+        # Should we strip out "special cases", i.e. service files, so that Heat can run unobstructed to create the bulk of the file and directory list ?
+        # Ruby erb templates have "directory list" that we can iterate over, so a Wix template with directories might work.
+        # Light command to generate msi
+        ]
+      end
+
+      # Method to derive the msi (Windows Installer) package name for the project.
+      #
+      # @param project [Vanagon::Project] project to name
+      # @return [String] name of the windows package for this project
+      def msi_package_name(project)
+        # Decided to use native project version in hope msi versioning doesn't have same resrictions as nuget
+        "#{project.name}-#{@architecture}-#{project.version}.msi"
```

Comment:
Again, we should keep this as close to the current versioning scheme as we can. This should be `"#{project.name}-#{project.version}.#{project.release}-#{@architecture}.msi"`, which corresponds with what we already are shipping `puppet-agent-1.3.2-x86.msi`